azure/scope/machine.go

```diff
@@ -363,6 +374,36 @@ func (m *MachineScope) ProviderID() string {
 	return parsed.String()
 }
 
+// AvailabilitySet returns the availability set for this machine if available.
+func (m *MachineScope) AvailabilitySetSpec() azure.ResourceSpecGetter {
+	availabilitySetName, ok := m.AvailabilitySet()
+	if !ok {
+		return nil
+	}
+
+	spec := &availabilitysets.AvailabilitySetSpec{
+		Name:             availabilitySetName,
+		ResourceGroup:    m.ResourceGroup(),
+		ClusterName:      m.ClusterName(),
+		Location:         m.Location(),
+		FaultDomainCount: nil,
+		AdditionalTags:   m.AdditionalTags(),
+	}
+
+	if m.cache != nil {
+		asSKU := m.cache.availabilitySetSKU
+		faultDomainCountStr, ok := asSKU.GetCapability(resourceskus.MaximumPlatformFaultDomainCount)
+		if ok {
+			faultDomainCount, err := strconv.ParseUint(faultDomainCountStr, 10, 32)
+			if err == nil {
```

Comment:
how about just putting `SKU` in `AvailabilitySetSpec` and just doing 
```
if m.cache != nil {
    spec.SKU = m.cache.availabilitySetSKU
}
```

in here, that way we're not silencing the actual error from getting the fault domain count if there is one? 

Then in the spec.go file you can use the SKU to get the fault domain count and return an error if appropriate.